### ISSUE TITLE
Add continuous integration

### ISSUE THREAD

**Issue by sidkshatriya** [Lullabot/amp-library#86]
Add continuous integration. 

Most probably we can use TravisCI as its hosted and works well with composer and phpunit

**Comment by sidkshatriya** [comment]
Travis CI allows testing with [`hhvm`](http://hhvm.com/) as a target. HHVM is being used by many big companies as a drop in replacement for PHP in many cases. It would be good to have our library be able to work with HHVM.

(Our library dependencies like `symfony`, `phpunit`, `html5-php` indicate that they are compatible with HHVM. Others dependencies should be OK too: HHVM and PHP 5.5+ compatibility is very good as I understand)

I ran our test suite in HHVM to see how we did.

```
$ hhvm vendor/bin/phpunit tests
PHPUnit 4.8.26 by Sebastian Bergmann and contributors.

F........................................

Time: 2.79 minutes, Memory: 60.76MB

There was 1 failure:

1) AmpTest::testFiles with data set "tests/test-data/fragment-html/audio-to-amp-audio-conversion-fragment.html" ('tests/test-data/fragment-html...t.html', false)
Failed asserting that two strings are equal.
```

...etc...

```
FAILURES!
Tests: 41, Assertions: 81, Failures: 1.
```

There is **only 1 failure**. This is great!

I looked into this failure and this is due to differences in `parse_url()` function for protocol-relative urls.
I've filed a bug report on the HHVM project. See https://github.com/facebook/hhvm/issues/7136

**Comment by sidkshatriya** [comment]
BTW don't be mislead by the test suite taking 2.79 minutes on HHVM (!!). Thats because of the JIT.

If you run the test with the jit disabled:

``` bash
$ hhvm -d hhvm.jit=false vendor/bin/phpunit tests
```

It will run in something more reasonable like 1.3 seconds or so...

**Comment by sidkshatriya** [comment]
I now have travis CI setup for continuous integration. Sadly I've not been able to set it up for the `lullabot/amp-library` repo... rather I've had to fork the repo to my own account as Travis says that I don't have the requisite rights (@dsayswhat FYI) 

Check the build status:
https://travis-ci.org/sidkshatriya/amp-library

Specifically you can check https://travis-ci.org/sidkshatriya/amp-library/builds/135871678

**Currently all the tests are succeeding for PHP 7.0** (See https://travis-ci.org/sidkshatriya/amp-library/jobs/135871682 )

![screenshot](https://cloud.githubusercontent.com/assets/638477/15859823/434d68fe-2ce4-11e6-9e9a-96baf589c588.png)

For PHP 5.5.21 and PHP 5.6.5 there is only 1 failure. See https://travis-ci.org/sidkshatriya/amp-library/jobs/135871681

And surprise surprise, **this is the same test that was failing for hhvm** (see https://github.com/Lullabot/amp-library/issues/86#issuecomment-224238454 above).  On my local machine (which has no failures) I'm running 5.5.33 and 5.6.19. So my guess is something changed for `parse_url()` between `5.5.21->5.5.33` and `5.6.5->5.6.19`

**Comment by sidkshatriya** [comment]
`parse_url()` problem has been confirmed as a PHP behaviour change on the HHVM ticket https://github.com/facebook/hhvm/issues/7136

Please see https://3v4l.org/fotNt (from https://github.com/facebook/hhvm/issues/7136#issuecomment-224427428) for a nice comparison

**Comment by sidkshatriya** [comment]
A workaround to the `parse_url()` problem was implemented in PR #90. All tests now pass in PHP 5.5, 5.6. 7.0, HHVM, PHP nightly. It does not matter if we have an older version of PHP 5.5 and 5.6 or HHVM (which is what the workaround fixes).

Travis CI is configured to test on PHP 5.5, 5.6, 7.0, HHVM and PHP nightly on every branch push. 

We also have a nice badge that indicates the build status.

![screenshot](https://cloud.githubusercontent.com/assets/638477/15888793/05907464-2d86-11e6-90b4-ab9a4e187d2e.png)

This ticket can be closed.
